dashboard.el

```diff
@@ -388,66 +388,52 @@ Optional argument ARGS adviced function arguments."
         (max-line-length 0))
     (when recentf-is-on
       (setq recentf-list (dashboard-subseq recentf-list dashboard-num-recents)))
-    (when (or dashboard-force-refresh
-              (not (eq dashboard-buffer-last-width (window-width))))
-      (setq dashboard-banner-length (window-width)
-            dashboard-buffer-last-width dashboard-banner-length)
-      (with-current-buffer (get-buffer-create dashboard-buffer-name)
-        (let (buffer-read-only)
-          (erase-buffer)
-          (dashboard-insert-banner)
-          (dashboard-insert-page-break)
-          (setq dashboard--section-starts nil)
-          (mapc (lambda (els)
-                  (let* ((el (or (car-safe els) els))
-                         (list-size
-                          (or (cdr-safe els)
-                              dashboard-items-default-length))
-                         (item-generator
-                          (cdr-safe (assoc el dashboard-item-generators))))
-                    (add-to-list 'dashboard--section-starts (point))
-                    (funcall item-generator list-size)
-                    (when recentf-is-on
-                      (setq recentf-list origial-recentf-list))
-                    (setq max-line-length
-                          (max max-line-length (dashboard-maximum-section-length)))
-                    (dashboard-insert-page-break)))
-                dashboard-items)
-          (when dashboard-center-content
-            (when dashboard--section-starts
-              (goto-char (car (last dashboard--section-starts))))
-            (let ((margin (floor (/ (max (- (window-width) max-line-length) 0) 2))))
-              (while (not (eobp))
-                (unless (string-suffix-p (thing-at-point 'line) dashboard-page-separator)
-                  (insert (make-string margin ?\ )))
-                (forward-line 1))))
-          (dashboard-insert-footer))
-        (goto-char (point-min))
-        (dashboard-mode)))
-    (when recentf-is-on
-      (setq recentf-list origial-recentf-list))))
-
-(add-hook 'window-setup-hook
-          (lambda ()
-            ;; 100 means `dashboard-resize-on-hook' will run last
-            (add-hook 'window-size-change-functions 'dashboard-resize-on-hook 100)
-            (dashboard-resize-on-hook)))
+    (prog1
+        (with-current-buffer (get-buffer-create dashboard-buffer-name)
+          (when (or dashboard-force-refresh (not (eq major-mode 'dashboard-mode)))
+            (let (buffer-read-only)
+              (erase-buffer)
+              (dashboard-insert-banner)
+              (setq dashboard--section-starts nil)
+              (mapc (lambda (els)
```

Comment:
Maybe we can named this `lambda` to `dashboard-items-list` or something more descriptive, I think is an important function and deserves a name 😄

Reply:
This lambda is currently a closure that reads/modifies the outer state.

Reply:
Ok, let's leave it as a "local" definition. But my point is that if a closure needs to read from the external state, it is better to have that value as a parameter and try to avoid "side effects" by not modifying the external scope.

Reply:
I generally agree (and didn't write this code, just moved it).